src/coreclr/src/gc/gc.cpp

```diff
@@ -33575,27 +35843,48 @@ size_t gc_heap::decommit_ephemeral_segment_pages_step ()
 #endif //MULTIPLE_HEAPS
 
 //This is meant to be called by decide_on_compacting.
-
 size_t gc_heap::generation_fragmentation (generation* gen,
                                           generation* consing_gen,
                                           uint8_t* end)
 {
-    size_t frag;
+    ptrdiff_t frag = 0;
```

Comment:
Is this really signed or unsigned? I see we compute it as a sum of differences of pointers, so ptrdiff_t is appropriate, but then we return it as a size_t, so we should probably add an assert to make sure it's >= 0.

Reply:
this is a bit involved... let me see if I can find you online to explain more. basically there was a perf bug in the old code (which only affects a certain scenario and doesn't cause a big problem so I just left the code as is - we can fix it but I didn't want to fix it in this PR 'cause I really tried to not change functionally for segments.